### PR TITLE
Logging Paradigm Revamp

### PR DESCRIPTION
**Pull request by n-beckage**
There are two major changes in this pull that address issue #12 :
1.  Updated `lib/get_calling_package()` so that it will return the package name of the original caller in the stack, as intended
2.  Added a class `lib/StreamToLogger` that when implemented, will forward all messages detined for the standard output (such as `print` statements) to the logger.
      - **Note**: `print` statements will not print to cosole _and_ the logger, but are now recorded in _just_ the logger. Since our logging setup has a `StreamHandler` as well, these messages do ultimately end up in the console, just as log messages rather than plain print statements

### CODE REVIEW

**Comment by pclemins** [comment]
Hey Noah... Mostly looks good.  2 questions:

1) in the logger commit, in lib.py, do we still need the log_name = calling_package.split('.')[-1] line before the one you added? (around line 194)
2) What happened to that testing function you showed me that printed out all the details about a frame.  I'd love to include that in this merge as well.

**Comment by pclemins** [comment]
Oh... and one more.  Could you add a similar comment like you did for the StreamToLogger for the logger = getLogger(...) line in lib.py.

File: lib.py [hunk 1]
```diff
@@ -190,6 +191,7 @@ def setup_logging(cls, name=None, default_level=logging.INFO):
 		calling_package = get_calling_package()
 		if(calling_package is not None):
 			log_name = calling_package.split('.')[-1]
```
Comment (pclemins) [inline lib.py:193]: Do we need this line now that you added the next one?

File: lib.py [hunk 2]
```diff
@@ -322,3 +355,5 @@ def multithreaded_loading(load_func, file_list, num_threads=int(os.cpu_count()/2
 
 IAMLogger.setup_logging()
 logger = logging.getLogger(get_calling_package())
```
Comment (pclemins) [inline lib.py:357]: Could you add a comment for this line like you did for StreamToLogger?

File: lib.py [hunk 3]
```diff
@@ -253,6 +272,20 @@ def get_calling_package():
 	calling_package = current_frame.f_globals['__package__']
 	# print(f'Calling Package: {calling_package}')
 	return calling_package
+"""
+
+# New get_calling_package()
```
Comment (pclemins) [inline lib.py:277]: What happened to that test/debug function that printed out all the details about a given frame?  I'd like to include that as well.

**Review by pclemins — CHANGES_REQUESTED** [review]

File: lib.py (hunk 1 above)
Comment (n-beckage) [reply lib.py:193]: Nope! will get rid of it.

File: lib.py (hunk 2 above)
Comment (n-beckage) [reply lib.py:357]: Yup!

**Pull request merged**